test-suite/CMakeLists.txt

```diff
@@ -180,3 +188,47 @@ if(DJINNI_WITH_OBJC)
 
   xctest_add_test(XCTest.DjinniObjcTest DjinniObjcTestTests) 
 endif()
+
+if(DJINNI_WITH_CPPCLI)
+  add_library(DjinniCppCli SHARED
+    ${CPP_HANDWRITTEN_SRCS}
+    ${CPP_GENERATED_SRCS}
+    ${CPPCLI_GENERATED_SRCS}
+    ${WCHAR_CPP_GENERATED_SRCS}
+    ${WCHAR_CPPCLI_GENERATED_SRCS}
+  )
+  target_include_directories(DjinniCppCli PUBLIC
+    ${CMAKE_CURRENT_SOURCE_DIR}/handwritten-src/cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/handwritten-src
+    ${CMAKE_CURRENT_BINARY_DIR}/generated-src/cpp
+    ${CMAKE_CURRENT_BINARY_DIR}/generated-src
+  )
+  set_target_properties(DjinniCppCli PROPERTIES
+    VS_DOTNET_REFERENCES "System;System.Core"
+    COMMON_LANGUAGE_RUNTIME ""
+  )
+  target_link_libraries(DjinniCppCli djinni_support_lib)
+
+  add_library(DjinniCppCliTest SHARED
+    handwritten-src/cs/ClientInterfaceImpl.cs
+    handwritten-src/cs/ClientInterfaceTest.cs
+    handwritten-src/cs/CppExceptionTest.cs
+    handwritten-src/cs/DurationTest.cs
+    handwritten-src/cs/EnumTest.cs
+    handwritten-src/cs/MapRecordTest.cs
+    handwritten-src/cs/MockRecordTest.cs
+    handwritten-src/cs/NestedCollectionTest.cs
+    handwritten-src/cs/PrimitiveListTest.cs
+    handwritten-src/cs/PrimitivesTest.cs
+    handwritten-src/cs/RecordWithDerivingsTest.cs
+    handwritten-src/cs/SetRecordTest.cs
+    handwritten-src/cs/TokenTest.cs
+    handwritten-src/cs/WcharTest.cs
+  )
+  set_target_properties(DjinniCppCliTest PROPERTIES
+    VS_DOTNET_REFERENCES "System;System.Core"
+    VS_PACKAGE_REFERENCES "nunit_3.13.1;NUnit3TestAdapter_3.17.0;Microsoft.NET.Test.Sdk_16.9.4"
+  )
```

Comment:
since cmake 3.15 even nuget dependencies can be added. CMake is awesome!
The only drawback is that they will not automatically be installed. If one would just configure the visual studio solution with cmake and then try to run `cmake --build`, this would fail because the nuget dependencies cannot be found. 
The workaround is to install them with the `dotnet` cli (`dotnet restore DjinniCppCliTest.csproj --runtime win-x64`) , or to open the solution in Visual Studio, which will automatically load the dependencies.

In had some trouble using the `dotnet` CLI, so I recommend to always build & run the solution in Visual Studio if you want to run it manually.

Reply:
Do you think it's still possible to run the tests as GitHub actions?

Reply:
I managed to build from the commandline on my local machine and I have created a github action that does the same steps. See the `build.yml` for details. If the `dotnet` CLI is available on the github runner, I am confident that it will work.